package/gluon-mesh-batman-adv/files/lib/netifd/proto/gluon_bat0.sh

```diff
@@ -29,10 +29,14 @@ proto_gluon_bat0_renew() {
 
 	lock /var/lock/gluon_bat0.lock
 
-	ubus call network.interface dump | jsonfilter \
-		-e "@.interface[@.proto='gluon_mesh' && @.up=true].device" \
-	| xargs -r -n 1 batctl interface add
-
+	for device in $(ubus call network.interface dump | jsonfilter -e "@.interface[@.proto='gluon_mesh' && @.up=true].device"); do
```

Comment:
If we can rely on every `gluon_mesh` interface having a `hop_penalty` set (and I believe we can, because `gluon_mesh.sh` always sets it), this can be handled much more efficiently with a single ubus and jsonfilter:
```sh
ubus call network.interface dump | \
    jsonfilter -e "@.interface[@.proto='static' && @.up=true]['device', 'hop_penalty']" | \
    while read device
do
    read hop_penalty
    ...
done
```

Reply:
sounds good - I did it that way